homeassistant/components/ezviz/alarm_control_panel.py

```diff
@@ -0,0 +1,165 @@
+"""Support for Ezviz alarm."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import timedelta
+import logging
+
+from pyezviz import PyEzvizError
+from pyezviz.constants import DefenseModeType
+
+from homeassistant.components.alarm_control_panel import (
+    AlarmControlPanelEntity,
+    AlarmControlPanelEntityDescription,
+    AlarmControlPanelEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_DISARMED,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    DATA_COORDINATOR,
+    DOMAIN,
+    MANUFACTURER,
+)
+from .coordinator import EzvizDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=60)
+PARALLEL_UPDATES = 0
+
+
+@dataclass
+class EzvizAlarmControlPanelEntityDescriptionMixin:
+    """Mixin values for EZVIZ Alarm control panel entities."""
+
+    ezviz_alarm_states: list
+
+
+@dataclass
+class EzvizAlarmControlPanelEntityDescription(
+    AlarmControlPanelEntityDescription, EzvizAlarmControlPanelEntityDescriptionMixin
+):
+    """Describe an EZVIZ Alarm control panel entity."""
+
+
+ALARM_TYPE = EzvizAlarmControlPanelEntityDescription(
+    key="ezviz_alarm",
+    ezviz_alarm_states=[
+        None,
+        STATE_ALARM_DISARMED,
+        STATE_ALARM_ARMED_AWAY,
+        STATE_ALARM_ARMED_HOME,
+    ],
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Ezviz alarm control panel."""
+    coordinator: EzvizDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        DATA_COORDINATOR
+    ]
+
+    device_info: DeviceInfo = {
+        "identifiers": {(DOMAIN, entry.unique_id)},  # type: ignore[arg-type]
+        "name": "EZVIZ Alarm",
```

Comment:
For some reason entry.unique_id is set to str | None

Is there a better way to do this? Also, though it might be cleaner to define device info here.

Reply:
`f"{entry.entry_id}_alarm"` would work aswell. Not every integration has unique_id set iirc.

Reply:
Understood. I initially thought elupus meant the config entry unique id.

Reply:
Depends on your integration, I think both work well. Only some cases it makes a difference (think of integrations that set the mac address as unique id, if someone removes the device and adds it again, it uses the same device identifiers). But I don't think that's a big deal here.

Reply:
EZVIZ uses the account email address (can't have more than one account on a specific email address)

I'll leave as is then.